src/components/application_manager/src/commands/hmi/vi_is_ready_request.cc

```diff
@@ -31,23 +31,69 @@
  */
 
 #include "application_manager/commands/hmi/vi_is_ready_request.h"
+#include "application_manager/message_helper.h"
 
 namespace application_manager {
 
 namespace commands {
 
 VIIsReadyRequest::VIIsReadyRequest(const MessageSharedPtr& message,
                                    ApplicationManager& application_manager)
-    : RequestToHMI(message, application_manager) {}
+    : RequestToHMI(message, application_manager)
+    , EventObserver(application_manager.event_dispatcher()){}
 
 VIIsReadyRequest::~VIIsReadyRequest() {}
 
 void VIIsReadyRequest::Run() {
   LOG4CXX_AUTO_TRACE(logger_);
-
+  subscribe_on_event(hmi_apis::FunctionID::VehicleInfo_IsReady,
+                     correlation_id());
   SendRequest();
 }
 
+void VIIsReadyRequest::on_event(const event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const smart_objects::SmartObject& message = event.smart_object();
+  switch (event.id()) {
+    case hmi_apis::FunctionID::VehicleInfo_IsReady: {
+      LOG4CXX_DEBUG(logger_, "VehicleInfo_IsReady event");
+      unsubscribe_from_event(hmi_apis::FunctionID::VehicleInfo_IsReady);
+      bool is_available = ChangeInterfaceState(
+                    application_manager_, message,
+                    HmiInterfaces::HMI_INTERFACE_VehicleInfo);
+
+      HMICapabilities& hmi_capabilities = application_manager_.hmi_capabilities();
+      hmi_capabilities.set_is_ivi_cooperating(is_available);
+      application_manager_.GetPolicyHandler().OnVIIsReady();
+      if(!CheckAvailabilityHMIInterfaces(application_manager_,
+                                         HmiInterfaces::HMI_INTERFACE_VehicleInfo)) {
+          LOG4CXX_INFO(logger_, "HmiInterfaces::HMI_INTERFACE_VehicleInfo isn't available");
+          return;
+      }
+      SendMessageToHMI();
+
+      break;
+  }
+    default: {
+      LOG4CXX_ERROR(logger_, "Received unknown event" << event.id());
+      return;
+    }
+  }
+}
+
+void VIIsReadyRequest::onTimeOut() {
+    //According wiht new requirment APPLINK-27958
```

Comment:
@dtrunov same here.. and please check everywhere else in order not to spam your PR about this in other files.